konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/serde/DataJsonSerializer.java

```diff
@@ -177,33 +178,50 @@ private void writeImage(JsonGenerator jg, Image i) throws IOException {
     }
 
     private void writeNDArray(JsonGenerator jg, NDArray n) throws IOException {
-        jg.writeStartObject();
+        if(n.rank() > 0 && n.rank() < 5) {
+            switch (n.rank()) {
```

Comment:
Maybe you could have reused the NDArrayStrings class in dl4j? It outputs valid json already.

Reply:
I can't use it here since it's part of the `nd4j-api` library and by design it's not supposed to be a part of `konduit-serving-pipeline`